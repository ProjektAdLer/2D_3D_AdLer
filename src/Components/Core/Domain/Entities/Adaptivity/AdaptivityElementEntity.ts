import { AdaptivityElementQuestionDifficultyTypes } from "../../Types/Adaptivity/AdaptivityElementQuestionDifficultyTypes";
import LearningElementEntity from "../LearningElementEntity";

export default class AdaptivityElementEntity {
  element: LearningElementEntity;
  introText: string;
  shuffleTask: boolean;
  tasks: AdaptivityTaskEntity[];
}

export class AdaptivityTaskEntity {
  taskId: number;
  taskTitle: string;
  taskOptional: boolean;
  requieredDifficulty: AdaptivityElementQuestionDifficultyTypes;
  questions: AdaptivityQuestionEntity[];
}

export class AdaptivityQuestionEntity {
  questionType: string;
  questionId: number;
  questionDifficulty: AdaptivityElementQuestionDifficultyTypes;
  questionText: string;
  questionAnswers: AdaptivityAnswerEntity[];
  trigger: AdaptivityTriggerEntity[];
}

export class AdaptivityAnswerEntity {
  answerId: number;
  answerText?: string;
  answerImage?: string;
}

export class AdaptivityTriggerEntity {
  triggerId: number;
  triggerType: string;
  triggerCondition: string;
  triggerAction: AdaptivityActionEntity;
}

export class AdaptivityActionEntity {
  actionType: string;
  actionData: string | number;
}
