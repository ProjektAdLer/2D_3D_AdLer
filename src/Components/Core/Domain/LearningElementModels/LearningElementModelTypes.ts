enum TextElementModelTypes {
  //default
  Bookshelf1 = "l_text_bookshelf_1",
  Bookshelf2 = "l_text_bookshelf_2",
  //arcade
  ComicShelfbig = "l_text_comicshelfbig_1",
  ComicShelfsmall = "l_text_comicshelfsmall_1",
  //campus
  LibraryShelf = "l_text_libraryshelf_1",
}

enum ImageElementModelTypes {
  //default
  Painting1 = "l_image_painting_1",
  Painting2 = "l_image_painting_2",
  PaintingEasel = "l_image_paintingeasel_1",
  //arcade
  Gameposter1 = "l_image_gameposter_1",
  Gameposter2 = "l_image_gameposter_2",
  Cardboardcutout = "l_image_cardboardcutout_1",
  //campus
  sciencegeo1 = "l_image_sciencegeo_1",
  sciencebio1 = "l_image_sciencebio_1",
  sciencewhiteboard = "l_image_sciencewhiteboard_1",
}

enum VideoElementModelTypes {
  //default
  Television = "l_video_television_1",
  //arcade
  vrdesk = "l_video_vrdesk_1",
  //campus
  movieprojector = "l_video_movieprojector_1",
}

// enum PdfElementModelTypes {}

enum H5pElementModelTypes {
  //default
  Blackboard = "l_h5p_blackboard_1",
  DeskPC1 = "l_h5p_deskpc_1",
  DrawingTable = "l_h5p_drawingtable_1",
  SlotMachine = "l_h5p_slotmachine_1",
  //arcade
  GreySlotmachine = "l_h5p_greyslotmachine_1",
  RedSlotmachine = "l_h5p_redslotmachine_1",
  PurpleSlotmachine = "l_h5p_purpleslotmachine_1",
  BlackSlotmachine = "l_h5p_blackslotmachine_1",
  DeskPC2 = "l_h5p_deskpc_2",
  //campus
  Blackboard2 = "l_h5p_blackboard_2",
  DeskPC3 = "l_h5p_deskpc_3",
  DrawingTable2 = "l_h5p_drawingtable_2",
  Daylightprojector = "l_h5p_daylightprojector_1",
}

enum QuizElementModelTypes {
  RobotNPC = "a_npc_alerobot",
  DefaultNPC = "a_npc_defaultnpc",
  ArcadeNPC = "a_npc_sheriffjustice",
  CampusNPC = "a_npc_dozentlukas",
}
enum TrophyElementModelTypes {
  BronzeTrophy = "l_trophy_bronze",
  SilverTrophy = "l_trophy_silver",
  GoldTrophy = "l_trophy_gold",
}

// enum NoElementModelTypes {
//   None = "",
// }

/** 
 Pseudo-nested enum to reference all possible members of the element model enums.
 */
export const LearningElementModelTypeEnums = {
  TextElementModelTypes: TextElementModelTypes,
  ImageElementModelTypes: ImageElementModelTypes,
  VideoElementModelTypes: VideoElementModelTypes,
  // PdfElementModelTypes: PdfElementModelTypes,
  H5pElementModelTypes: H5pElementModelTypes,
  QuizElementModelTypes: QuizElementModelTypes,
  TrophyElementModelTypes: TrophyElementModelTypes,
  // NoElementModelTypes: NoElementModelTypes,
};

export function isValidLearningElementModelType(type: string): boolean {
  return Object.values(LearningElementModelTypeEnums).some((enumType) =>
    Object.values(enumType).includes(type)
  );
}

/**
 Type of a learning element model.
  */
export type LearningElementModel =
  | TextElementModelTypes
  | ImageElementModelTypes
  | VideoElementModelTypes
  // | PdfElementModelTypes
  | H5pElementModelTypes
  | QuizElementModelTypes
  | TrophyElementModelTypes;
// | NoElementModelTypes;
