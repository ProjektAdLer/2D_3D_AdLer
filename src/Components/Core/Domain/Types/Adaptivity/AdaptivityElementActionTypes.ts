// TODO: add string representations
export enum AdaptivityElementActionTypes {
  CommentAction = "CommentAction",
  ReferenceAction = "AdaptivityReferenceAction",
  ContentAction = "AdaptivityContentReferenceAction",
}
