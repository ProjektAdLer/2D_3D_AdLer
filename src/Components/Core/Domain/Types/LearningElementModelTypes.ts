enum TextElementModelTypes {
  Bookshelf1 = "l_text_bookshelf_1",
  Bookshelf2 = "l_text_bookshelf_2",
}

enum ImageElementModelTypes {
  // TODO: rename to l_image_... when models are updated
  Painting1 = "l_picture_painting_1",
  Painting2 = "l_picture_painting_2",
  PaintingEasel = "l_picture_paintingeasel_1",
}

enum VideoElementModelTypes {
  Television = "l_video_television_1",
}

enum PdfElementModelTypes {}

enum H5pElementModelTypes {
  Backboard = "l_h5p_backboard_1",
  DeskPC = "l_h5p_deskpc_1",
  DrawingTable = "l_h5p_drawingtable_1",
  SlotMachine = "l_h5p_slotmachine_1",
}

enum NoElementModelTypes {
  None = "",
}

/** 
 Pseudo-nested enum to reference all possible members of the element model enums.
 */
export const LearningElementModelTypeEnums = {
  TextElementModelTypes: TextElementModelTypes,
  ImageElementModelTypes: ImageElementModelTypes,
  VideoElementModelTypes: VideoElementModelTypes,
  PdfElementModelTypes: PdfElementModelTypes,
  H5pElementModelTypes: H5pElementModelTypes,
  NoElementModelTypes: NoElementModelTypes,
};

/**
 Type of a learning element model.
  */
export type LearningElementModel =
  | TextElementModelTypes
  | ImageElementModelTypes
  | VideoElementModelTypes
  | PdfElementModelTypes
  | H5pElementModelTypes
  | NoElementModelTypes;
