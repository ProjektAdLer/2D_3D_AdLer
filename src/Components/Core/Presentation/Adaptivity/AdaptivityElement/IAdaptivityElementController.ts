export default interface IAdaptivityElementController {
  closeModal(): void;
  selectTask(taskID: number): void;
  selectQuestion(selectedQuestionID: number): void;
  submitSelection(): void;
  closeFeedback(): void;
  closeAnswerSelection(): void;
  back(): void;
}
