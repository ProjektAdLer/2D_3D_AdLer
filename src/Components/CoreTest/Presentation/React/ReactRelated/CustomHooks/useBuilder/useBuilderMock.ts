import * as useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";

/**
 * Mocks the useViewModelControllerProvider hook.
 * @param mockedTupels The return value, which the mocked hook will return.
 *
 */
export default function useBuilderMock(
  mockedTupels: [viewModel: any, controller: any] = [undefined, undefined]
) {
  jest.spyOn(useBuilder, "default").mockReturnValue(mockedTupels);
}
