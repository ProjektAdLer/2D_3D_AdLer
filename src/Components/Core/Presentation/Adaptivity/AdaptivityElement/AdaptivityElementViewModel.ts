import { AdaptivityElementQuestionDifficultyTypes } from "src/Components/Core/Domain/Types/Adaptivity/AdaptivityElementQuestionDifficultyTypes";
import Observable from "src/Lib/Observable";
import { StyledButtonColor } from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";

export interface AdaptivityAnswer {
  answerIndex: number;
  answerText: string;
  isSelected: boolean;
}

export interface AdaptivityQuestion {
  questionID: number;
  questionText: string;
  questionAnswers: AdaptivityAnswer[];
  isRequired: boolean;
  isCompleted: boolean;
  difficulty: AdaptivityElementQuestionDifficultyTypes;
  isMultipleChoice: boolean;
}

export interface AdaptivityTask {
  taskID: number;
  taskTitle: string;
  questions: AdaptivityQuestion[];
  isCompleted: boolean;
  requiredDifficulty: number;
}

export interface AdaptivityElementContent {
  elementName: string;
  introText: string;
  tasks: AdaptivityTask[];
}

export default class AdaptivityElementViewModel {
  isOpen: Observable<boolean> = new Observable<boolean>(false);
  showFeedback: Observable<boolean> = new Observable<boolean>(false);
  contentData: Observable<AdaptivityElementContent> =
    new Observable<AdaptivityElementContent>();
  currentTaskID: Observable<number | null> = new Observable<number | null>(
    null
  );
  currentQuestionID: Observable<number | null> = new Observable<number | null>(
    null
  );
  footerText: Observable<string> = new Observable<string>("");
}
