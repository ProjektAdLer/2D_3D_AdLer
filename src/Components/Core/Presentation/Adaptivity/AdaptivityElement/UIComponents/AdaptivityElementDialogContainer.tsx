import { useState } from "react";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import AdaptivityElementViewModel from "../AdaptivityElementViewModel";
import IAdaptivityElementController from "../IAdaptivityElementController";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import useObservable from "~ReactComponents/ReactRelated/CustomHooks/useObservable";
import StyledContainer from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledContainer";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";
import tailwindMerge from "../../../Utils/TailwindMerge";
import { AdLerUIComponent } from "../../../../Types/ReactTypes";

import quizBackgroundVRGuy from "../../../../../../Assets/misc/quizBackgrounds/vr-guy-quiz-background.png";
import AdaptivityElementTaskSelection from "./AdaptivityElementTaskSelection";

export default function AdaptivityElementDialogContainer({
  className,
}: AdLerUIComponent) {
  const [viewmodel, controller] = useBuilder<
    AdaptivityElementViewModel,
    IAdaptivityElementController
  >(BUILDER_TYPES.IAdaptivityElementBuilder);

  const [isOpen] = useObservable<boolean>(viewmodel?.isOpen);
  const [contentData] = useObservable(viewmodel?.contentData);

  // TODO: remove DEBUG data
  const [footer] = useState("Adaptivit√§tselement");

  if (!viewmodel || !controller) return null;
  if (!isOpen || !contentData) return null;

  return (
    <>
      <img
        className="absolute top-0 right-0 z-10 max-h-[93vh] w-fit max-w-[90vw] lg:max-w-[99vw]"
        alt="LearningImage!"
        src={quizBackgroundVRGuy}
      ></img>
      <StyledContainer className={tailwindMerge(className, "")}>
        <div className="fixed top-0 bottom-0 left-0 right-0 flex flex-col items-center justify-center h-full">
          <div className="p-2 rounded-lg bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto max-w-[95%] max-h-[95%] overflow-auto">
            <div className="flex items-center justify-center w-full gap-2 p-1 pb-3 text-xl font-bold text-adlerdarkblue lg:roboto-black lg:text-2xl h-fit">
              <div className="w-full">{contentData.elementName}</div>

              <StyledButton
                onClick={controller.closeModal}
                className="w-8 h-8 p-1 text-xs roboto-black xl:w-10 xl:h-10 lg:w-10 lg:h-10 md:w-10 md:h-10 sm:w-10 sm:h-10"
              >
                X
              </StyledButton>
            </div>

            <div className="px-1 rounded-lg font-regular h-fit">
              <AdaptivityElementTaskSelection
                tasks={contentData.tasks}
                onSelectTask={controller.selectTask}
              />
            </div>

            {
              <div className="modal-footer">
                <p>{footer}</p>
              </div>
            }
          </div>
        </div>
      </StyledContainer>
    </>
  );
}
