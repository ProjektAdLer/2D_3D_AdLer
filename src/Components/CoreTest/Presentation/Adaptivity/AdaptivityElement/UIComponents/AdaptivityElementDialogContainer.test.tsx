import { getAllByTestId, render, waitFor } from "@testing-library/react";
import React from "react";
import AdaptivityElementDialogContainer from "../../../../../Core/Presentation/Adaptivity/AdaptivityElement/UIComponents/AdaptivityElementDialogContainer";
import useBuilderMock from "../../../React/ReactRelated/CustomHooks/useBuilder/useBuilderMock";
import AdaptivityElementViewModel, {
  AdaptivityElementContent,
} from "../../../../../Core/Presentation/Adaptivity/AdaptivityElement/AdaptivityElementViewModel";
import { mock } from "jest-mock-extended";
import IAdaptivityElementController from "../../../../../Core/Presentation/Adaptivity/AdaptivityElement/IAdaptivityElementController";
import { AdaptivityElementActionTypes } from "../../../../../Core/Domain/Types/Adaptivity/AdaptivityElementActionTypes";
import { LearningElementModelTypeEnums } from "../../../../../Core/Domain/LearningElementModels/LearningElementModelTypes";

describe("AdaptivityElementDialogContainer", () => {
  let viewModel: AdaptivityElementViewModel;

  beforeEach(() => {
    viewModel = new AdaptivityElementViewModel();
    useBuilderMock([viewModel, mock<IAdaptivityElementController>()]);
  });

  test("doesn't render if viewModel is undefined", () => {
    useBuilderMock([undefined, undefined]);
    const { container } = render(<AdaptivityElementDialogContainer />);
    expect(container).toBeEmptyDOMElement();
  });

  test("doesn't render if controller is undefined", () => {
    useBuilderMock([viewModel, undefined]);
    const { container } = render(<AdaptivityElementDialogContainer />);
    expect(container).toBeEmptyDOMElement();
  });

  test("doesn't render if isOpen is false", () => {
    viewModel.isOpen.Value = false;

    const { container } = render(<AdaptivityElementDialogContainer />);
    expect(container).toBeEmptyDOMElement();
  });

  test("doesn't render if contentData is undefined", () => {
    viewModel.isOpen.Value = true;
    viewModel.contentData.Value = undefined;

    const { container } = render(<AdaptivityElementDialogContainer />);
    expect(container).toBeEmptyDOMElement();
  });

  // ANF-ID: [EWE0014]
  test("should render", () => {
    viewModel.isOpen.Value = true;
    viewModel.model.Value =
      LearningElementModelTypeEnums.H5pElementModelTypes.Blackboard;
    const question = {
      questionID: 0,
      questionText: "testQuestionText",
      isMultipleChoice: false,
      isRequired: false,
      isCompleted: false,
      difficulty: 0,
      questionAnswers: [
        {
          answerIndex: 0,
          answerText: "testAnswerText",
          isSelected: false,
        },
      ],
      hints: [
        {
          hintID: 0,
          hintAction: {
            hintActionData: "testHintActionData",
            hintActionType: AdaptivityElementActionTypes.CommentAction,
          },
          showOnIsWrong: false,
        },
      ],
    };
    const task = {
      taskID: 0,
      taskTitle: "testTaskTitle",
      isCompleted: false,
      isRequired: false,
      hasBeenCompleted: false,
      requiredDifficulty: 0,
      questions: [question],
    };

    const content: AdaptivityElementContent = {
      elementName: "testName",
      introText: "testIntroText",
      tasks: [task],
    };
    viewModel.contentData.Value = content;
    viewModel.currentTask.Value = task;
    viewModel.currentQuestion.Value = question;

    const { container } = render(<AdaptivityElementDialogContainer />);
    expect(container).toMatchSnapshot();
  });

  // ANF-ID: [EWE0015]
  test.each([
    [LearningElementModelTypeEnums.QuizElementModelTypes.ArcadeNPC],
    [LearningElementModelTypeEnums.QuizElementModelTypes.CampusNPC],
    [LearningElementModelTypeEnums.QuizElementModelTypes.DefaultNPC],
    [LearningElementModelTypeEnums.QuizElementModelTypes.RobotNPC],
  ])("displays correct npc thumbnail for model %p", (model) => {
    viewModel.isOpen.Value = true;
    viewModel.model.Value = model;
    const question = {
      questionID: 0,
      questionText: "testQuestionText",
      isMultipleChoice: false,
      isRequired: false,
      isCompleted: false,
      difficulty: 0,
      questionAnswers: [
        {
          answerIndex: 0,
          answerText: "testAnswerText",
          isSelected: false,
        },
      ],
      hints: [
        {
          hintID: 0,
          hintAction: {
            hintActionData: "testHintActionData",
            hintActionType: AdaptivityElementActionTypes.CommentAction,
          },
          showOnIsWrong: false,
        },
      ],
    };
    const task = {
      taskID: 0,
      taskTitle: "testTaskTitle",
      isCompleted: false,
      isRequired: false,
      hasBeenCompleted: false,
      requiredDifficulty: 0,
      questions: [question],
    };

    const content: AdaptivityElementContent = {
      elementName: "testName",
      introText: "testIntroText",
      tasks: [task],
    };
    viewModel.contentData.Value = content;
    viewModel.currentTask.Value = task;
    viewModel.currentQuestion.Value = question;

    const { container } = render(<AdaptivityElementDialogContainer />);

    const npcImages = getAllByTestId(container, "npcImage");

    expect(npcImages).toHaveLength(2);
    npcImages.forEach((npcImage) => {
      expect(npcImage).toBeInTheDocument();
      expect(npcImage).toMatchSnapshot();
    });
  });

  // ANF-ID: [EWE0019]
  test("click on legend-text calls showFooterTooltip", () => {
    const controllerMock = mock<IAdaptivityElementController>();

    const content: AdaptivityElementContent = {
      elementName: "testName",
      introText: "testIntroText",
      tasks: [],
    };
    viewModel.contentData.Value = content;
    viewModel.isOpen.Value = true;

    useBuilderMock([viewModel, controllerMock]);

    const { getByText } = render(<AdaptivityElementDialogContainer />);
    getByText("legendHover").click();
    expect(controllerMock.showFooterTooltip).toBeCalled();
  });

  // ANF-ID: [EWE0019]
  test("displays legend", () => {
    viewModel.showFooterTooltip.Value = true;
    const container = render(<AdaptivityElementDialogContainer />);

    waitFor(() => {
      expect(container.getByText("headerLegend")).toBeInTheDocument();
      expect(
        container.getByTestId("requiredTaskIconImage")
      ).toBeInTheDocument();
    });
  });
});
