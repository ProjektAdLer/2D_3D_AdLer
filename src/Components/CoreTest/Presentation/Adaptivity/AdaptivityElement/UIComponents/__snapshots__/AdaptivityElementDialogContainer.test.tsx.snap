// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptivityElementDialogContainer should render 1`] = `
<div>
  <div
    class=" flex gap-2 m-2 w-fit h-fit text-adlertextgrey"
  >
    <div
      class="fixed top-0 bottom-0 left-0 right-0 flex flex-col items-center justify-center w-screen h-screen bg-black bg-opacity-50 lg:grid lg:grid-rows-3 lg:items-start"
    >
      <div
        class="flex items-end justify-start invisible w-full lg:visible lg:h-full"
      >
        <img
          alt="LearningImage!"
          class="z-20 invisible object-contain h-0 pl-16 lg:visible lg:h-full "
          src="vr-guy-quiz-background_cutted.png"
        />
      </div>
      <div
        class="flex justify-center items-start lg:row-span-2 w-full lg:w-[95vw] max-w-7xl h-full pt-2 lg:pt-0"
      >
        <div
          class="flex flex-col p-2 xl:px-8 rounded-lg bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto w-full max-w-[95%] max-h-[95%] lg:h-fit justify-between overflow-auto"
        >
          <div
            class="z-20 flex items-center justify-center w-full gap-2 p-2 pb-3 text-xl font-bold text-adlerdarkblue lg:roboto-black lg:text-2xl "
          >
            <img
              alt="LearningImage!"
              class="visible h-16 lg:invisible lg:h-0"
              src="vr-guy-quiz-background.png"
            />
            <div
              class="w-full text-sm lg:text-xl"
            >
              testQuestionText
            </div>
            <button
              class="roboto-black xl:w-10 xl:h-10 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-1 active:translate-y-1 active:border-transparent text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border bg-buttonbgblue"
            >
              &lt;
            </button>
            <button
              class="roboto-black xl:w-10 xl:h-10 justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-1 active:translate-y-1 active:border-transparent text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border bg-buttonbgblue"
            >
              X
            </button>
          </div>
          <div
            class="flex items-center justify-center px-1 mb-4 rounded-lg font-regular h-fit lg:m-4"
          >
            <div
              class="flex flex-col w-full p-2 m-auto"
            >
              <div
                class="flex flex-col flex-wrap justify-start gap-4 lg:grid lg:grid-cols-2 "
              >
                <button
                  class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-1 active:translate-y-1 active:border-transparent text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border bg-buttonbgblue"
                >
                  testAnswerText
                </button>
              </div>
              <div
                class="flex items-end justify-end w-full gap-2 pt-8 font-bold"
              >
                <button
                  class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-1 active:translate-y-1 active:border-transparent text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border bg-buttonbgblue"
                >
                  Zur√ºck
                </button>
                <button
                  class="px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-1 active:translate-y-1 active:border-transparent text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border bg-buttonbgblue"
                >
                  Antworten abgeben
                </button>
              </div>
            </div>
          </div>
          <div
            class="text-xs modal-footer lg:text-sm"
          >
            <p />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
